Deprecate DependencyInsightReportTask.getDependencySpec (#31517)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

While explicit methods can be used with space, like `propertyName value`, consider using the `propertyName(value)` syntax for clarity.
For example, `jvmArgs "some", "arg"` can be replaced with `jvmArgs("some", "arg")` or with `jvmArgs = ["some", "arg"]` for `Test` tasks.

[[dependency-insight-report-task-get-dependency-spec]]
==== DependencyInsightReportTask.getDependencySpec

> The method was deprecated since it's not intended to be public and used by build scripts.

Reviewer: ```suggestion
The method was deprecated because it was not intended for public use in build scripts.
```
